modules/hal_rpi_pico/CMakeLists.txt

```diff
@@ -51,6 +51,7 @@ if(CONFIG_HAS_RPI_PICO)
   # initializing the SoC, and therefore are always required.
 
   zephyr_library_sources(
+    ${rp2_common_dir}/hardware_claim/claim.c
```

Comment:
`claim` is not always needed, it can be added conditionally the same way it is done in the PIO PR

Reply:
Thank you for reviewing.

I pick code from PIO PR and apply it.